tests/integration/test_scale_means.py

```diff
@@ -237,7 +242,18 @@ def test_cat_x_cat_pruning_and_hs():
     assert_scale_means_equal(actual, expected)
     expected = [[
         np.array([1.57933884, np.nan, 2.10618401, 2.30460074, 2.34680135]),
-        np.array([1.74213625, np.nan, 1.97      , 2.45356177, 2.11838791, 2.]),
+        np.array([1.74213625, np.nan, 1.97      , 2.45356177, 2.11838791, 2.]),  # noqa
     ]]
     actual = cube.scale_means(hs_dims=[0, 1], prune=True)
     assert_scale_means_equal(actual, expected)
+
+
+def test_cat_x_cat_scale_means_margin():
+    cs = CrunchCube(CAT_X_CAT_SM_MARGIN['slides'][0]['cube']).slices[0]
+    expected = 2.6846246973365617
+    actual = cs.scale_means_margin(1)
+    assert actual == expected
+
+    expected = 2.536319612590799
+    actual = cs.scale_means_margin(0)
+    assert actual == expected
```

Comment:
Are these the only new expectations? I scrolled through and it looked like most of the ones above were just `#noqa` additions, but I want to make sure that I'm not missing something!

Reply:
These values, however are good!

```
> cube <- loadCube("~/Desktop/cat-x-cat-sm-margin.json") # edited to extract just the cube
> 
> # get the category values
> categories(variables(cube)[[1]])
     id        name value missing
1     1    Bucatini     1   FALSE
2     2    Chitarra     2   FALSE
3     0     Boccoli     0   FALSE
4     4 Orecchiette     4   FALSE
5     5 Quadrefiore     5   FALSE
6     6      Fileja     6   FALSE
7 32766     Skipped 32766    TRUE
8 32767   Not asked 32767    TRUE
9    -1     No Data    NA    TRUE
> buc <- 1
> chi <- 2
> boc <- 0
> ore <- 4
> qua <- 5
> fil <- 6
> 
> # calcualte the univariate mean for rows
> sum(margin.table(cube, 1) * c(buc, chi, boc, ore, qua, fil)) / margin.table(cube)
[1] 2.684625
> 
> categories(variables(cube)[[2]])
     id       name value missing
1     0 Vegetables     0   FALSE
2 32766 Don't know 32766    TRUE
3     2      Fruit     2   FALSE
4     5      Grain     5   FALSE
5    -1    No Data    NA    TRUE
6     4       Meat     4   FALSE
7 32767  Not asked 32767    TRUE
> veg <- 0
> fruit <- 2
> grain <- 5
> meat <- 4
> 
> # calcualte the univariate mean for cols
> sum(margin.table(cube, 2) * c(veg, fruit, grain, meat)) / margin.table(cube)
[1] 2.53632
```

Reply:
Yup, these are the only expectations. We can add more when we do exporter tests. But I wanted to keep it as simple as possible for the first iteration.

Reply:
i'd rather have more tests here and less in exporter.  Try to keep the tests as close to the determination of values as possible.

Reply:
Done, I've updated the existing tests with calls to new functionality.